framework/PostProcessors/BasicStatistics.py

```diff
@@ -1045,91 +800,80 @@ def getCovarianceSubset(desired):
-      elif metric in self.scalarVals:
-        outputDict[metric] = dict((target,calculations[metric][target]) for target in params)
-      #if a matrix block, extract desired values
-      else:
-        if metric in ['pearson','covariance']:
-          outputDict[metric] = {}
-          for entry in params:
-            #check if it was skipped for some reason
-            if entry == self.skipped.get(metric,None):
-              self.raiseADebug('Metric',metric,'was skipped for parameters',entry,'!  See warnings for details.  Ignoring...')
-              continue
-            for target in entry['targets']:
-              if target not in outputDict[metric].keys():
-                outputDict[metric][target] = {}
-              targetIndex = calculations[metric]['params'].index(target)
-              for feature in entry['features']:
+        continue
+      for targetDict in requestList:
+        prefix = targetDict['prefix'].strip()
+        if metric in self.scalarVals:
+          for targetP in targetDict['targets']:
+            varName = prefix + '_' + targetP
+            outputDict[varName] = np.atleast_1d(calculations[metric][targetP])
+        #TODO someday we might need to expand the "skipped" check to include scalars, but for now
+        #   the only reason to skip is if an invalid matrix is requested
+        #if matrix block, extract desired values
+        else:
+          #check if it was skipped for some reason
+          skip = self.skipped.get(metric, None)
+          if skip is not None:
+            self.raiseADebug('Metric',metric,'was skipped for parameters',targetDict,'!  See warnings for details.  Ignoring...')
+            continue
+          if metric in ['pearson', 'covariance']:
+            for targetP in targetDict['targets']:
+              targetIndex = calculations[metric]['params'].index(targetP)
+              for feature in targetDict['features']:
+                varName = prefix + '_' + targetP + '_' + feature
                 featureIndex = calculations[metric]['params'].index(feature)
-                outputDict[metric][target][feature] = calculations[metric]['matrix'][targetIndex,featureIndex]
-        #if matrix but stored in dictionaries, just grab the values
-        elif metric in ['sensitivity','NormalizedSensitivity','VarianceDependentSensitivity']:
-          outputDict[metric] = {}
-          for entry in params:
-            #check if it was skipped for some reason
-            if entry == self.skipped.get(metric,None):
-              self.raiseADebug('Metric',metric,'was skipped for parameters',entry,'!  See warnings for details.  Ignoring...')
-              continue
-            for target in entry['targets']:
-              outputDict[metric][target] = dict((feature,calculations[metric][target][feature]) for feature in entry['features'])
+                outputDict[varName] = np.atleast_1d(calculations[metric]['matrix'][targetIndex,featureIndex])
```

Comment:
the multiple entries for a metric is available for jjust some relational metrics and not for others?

Reply:
Sorry, I did not get your point.